setuptools/command/bdist_wheel.py

```diff
@@ -295,21 +305,21 @@ def finalize_options(self):
             raise ValueError("Build tag (build-number) must start with a digit.")
 
     @property
-    def wheel_dist_name(self):
+    def wheel_dist_name(self) -> str:
         """Return distribution full name with - replaced with _"""
-        components = (
+        components = [
             safer_name(self.distribution.get_name()),
             safer_version(self.distribution.get_version()),
-        )
+        ]
         if self.build_number:
-            components += (self.build_number,)
+            components.append(self.build_number)
         return "-".join(components)
 
     def get_tag(self) -> tuple[str, str, str]:
         # bdist sets self.plat_name if unset, we should only use it for purepy
         # wheels if the user supplied it.
-        if self.plat_name_supplied:
-            plat_name = cast(str, self.plat_name)
+        if self.plat_name_supplied and self.plat_name:
```

Comment:
Umm... that is a bit intriguing isn't it.

It seems that `self.plat_name_supplied` is only used once in the whole file, so why do we need it to cache the result of such simple information like `if self.plat_name`?

@agronholm do you have any thoughts about this?

Reply:
All of `bdist_wheel` code has been yucky, and there are lots of things that don't make sense, so I've refrained from refactoring it myself. This is just one of those examples. I don't really see why this extra member is needed, so your suggestion makes sense.

Reply:
On further inspection, I see that the else statement do a separated check on `self.plat_name`.

So I suppose this variable distinguishes between "given by the user" and "programmatically set".

Reply:
> So I suppose this variable distinguishes between "given by the user" and "programmatically set".

That's what I figured looking at it too